pkgs/development/haskell-modules/generic-stack-builder.nix

```diff
@@ -33,7 +33,7 @@ in stdenv.mkDerivation (args // {
   STACK_PLATFORM_VARIANT="nix";
   STACK_IN_NIX_SHELL=1;
   STACK_IN_NIX_EXTRA_ARGS = extraArgs;
-  shellHook = addStackArgsHook;
+  shellHook = addStackArgsHook + args.shellHook;
```

Comment:
Will throw an error when `shellHook` isn't set, so you need `+ args.shellHook or ""`